### PR TITLE
better build menu

### PR DESCRIPTION
**Pull request by Bod9001**
GIF226

### Changelog:
CL: [Improvement] you can now select how many you want to build in the build menu for materials

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs [hunk 1]
```diff
@@ -94,7 +94,17 @@ public void OnClick()
 				return;
 			}
 
-			RequestBuildMessage.Send(entry, buildingMaterial);
+			int NumberWanted = 1;
+
+
+			if (int.TryParse(UIManager.BuildMenu.NumberInputField.text, out NumberWanted) == false)
+			{
+				NumberWanted = 1;
```
Comment (AtnerNT) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs:102]: This assignment of NumberWanted is unnecessary? Already set to one above

File: UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs [hunk 2]
```diff
@@ -76,47 +85,67 @@ public override void Process(NetMessage msg)
 				}
 			}
 
-			//build and consume
-			void ProgressComplete()
+			if (entry.OnePerTile)
```
Comment (AtnerNT) [inline UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs:88]: Not nessarcary, but could remove these brackets and put this on one line like:
if(entry.OnePerTile == true) msg.Number = 1;

Just to clean it up visually.

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs (hunk 1 above)
Comment (Bod9001) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs]: Just catch the out settings to a funny number?

File: UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs:88]: Cool will do

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs (hunk 1 above)
Comment (AtnerNT) [reply UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs]: Oh yeah misread

File: UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs (hunk 2 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs:88]: we agreed on only doing that when returning

File: UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs [hunk 3]
```diff
@@ -76,47 +85,67 @@ public override void Process(NetMessage msg)
...
+
 		/// <summary>
 		/// Request constructing the given entry
 		/// </summary>
 		/// <param name="entry">entry to build</param>
 		/// <param name="hasMenu">has construction menu component of the object being used to
 		/// construct.</param>
 		/// <returns></returns>
-		public static NetMessage Send(BuildList.Entry entry, BuildingMaterial hasMenu)
+		public static NetMessage Send(BuildList.Entry entry, BuildingMaterial hasMenu, int Number)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs]:
```suggestion
		public static NetMessage Send(BuildList.Entry entry, BuildingMaterial hasMenu, int number)
```

File: UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs [hunk 4]
```diff
@@ -76,47 +85,67 @@ public override void Process(NetMessage msg)
...
 		{
 			int entryIndex = hasMenu.BuildList.Entries.ToList().IndexOf(entry);
 			if (entryIndex == -1) return new NetMessage(); // entryIndex was previously a byte, which made this check impossible.
 
 			NetMessage msg = new NetMessage
 			{
-				EntryIndex = (byte) entryIndex
+				EntryIndex =  entryIndex,
+				Number =Number
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Messages/Client/RequestBuildMessage.cs]:
```suggestion
				Number = number
```

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenu.cs [hunk 5]
```diff
@@ -23,6 +24,10 @@ public class BuildMenu : MonoBehaviour
 		// current object whose menu is being shown
 		private BuildingMaterial currentBuildingMaterial;
 
+		[Tooltip("The number of the specified item to make ")]
+		[SerializeField] public TMP_InputField NumberInputField = null;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenu.cs:28]: why public if you added ``SerializeField``?

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs [hunk 6]
```diff
@@ -94,7 +94,17 @@ public void OnClick()
 				return;
 			}
 
-			RequestBuildMessage.Send(entry, buildingMaterial);
+			int NumberWanted = 1;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs]:
```suggestion
			int numberWanted = 1;
```

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs (hunk 1 above)
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs]:
```suggestion
				numberWanted = 1;
```

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs [hunk 7]
```diff
@@ -94,7 +94,17 @@ public void OnClick()
...
+
+
+			if (int.TryParse(UIManager.BuildMenu.NumberInputField.text, out NumberWanted) == false)
+			{
+				NumberWanted = 1;
+				UIManager.BuildMenu.NumberInputField.text = NumberWanted.ToString();
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs]:
```suggestion
				UIManager.BuildMenu.NumberInputField.text = numberWanted.ToString();
```

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs [hunk 8]
```diff
@@ -94,7 +94,17 @@ public void OnClick()
...
+			}
+
+
+			RequestBuildMessage.Send(entry, buildingMaterial, NumberWanted);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenuEntryController.cs]:
```suggestion
			RequestBuildMessage.Send(entry, buildingMaterial, numberWanted);
```

**Review by corp-0 — CHANGES_REQUESTED** [review]
you can commit the suggestions if you want.

File: UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenu.cs (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/Construction/BuildMenu.cs:28]: Was private but then needed to access

**Pull request merged**